crates/sui-types/src/layout_resolver.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    error::SuiError,
+    object::{MoveObject, ObjectFormatOptions},
+};
+use move_core_types::value::MoveStructLayout;
+
+pub trait LayoutResolver {
+    fn get_layout(
+        &mut self,
+        object: &MoveObject,
```

Comment:
nit: Should this accept the full object, or just its type?

Reply:
not sure maybe both API in time?

Reply:
leaving it as is for now